File: 2005/Windows SDK NetFx 3.0/Applications/CultureSample/VB/CultureSample/CultureSample.vb
 '-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Globalization


'/ <summary>
'/ Main window for the application.
'/ </summary>

Public Class CultureBuilder
    Inherits System.Windows.Forms.Form
    #Region "Windows Form Designer declarations"
    Private toolStripTop As System.Windows.Forms.ToolStrip
    Private WithEvents newToolStripButton As System.Windows.Forms.ToolStripButton
    Private WithEvents mixToolStripButton As System.Windows.Forms.ToolStripButton
    Private WithEvents instanceToolStripButton As System.Windows.Forms.ToolStripButton
    Private panelDisplayCulture As System.Windows.Forms.Panel
    #End Region
    
    '/ <summary>
    '/ Required designer variable.
    '/ </summary>
    Private components As System.ComponentModel.Container = Nothing
    
    'Helper to pass on to the children windows
    Private helper As CultureInfoHelper
    
    
    Public Sub New() 
        '
        ' Required for Windows Form Designer support
        '
        InitializeComponent()
        Dim display As New DisplayCulture()
        helper = New CultureInfoHelper(display)
        display.LoadComboBox(helper.GetCultures(CultureTypes.AllCultures))
        panelDisplayCulture.Controls.Add(display)
    
    End Sub 'New
    
    
    '/ <summary>
    '/ Clean up any resources being used.
    '/ </summary>
    Protected Overrides Sub Dispose(ByVal disposing As Boolean) 
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    
    End Sub 'Dispose
    
    #Region "Windows Form Designer generated code"
    
    '/ <summary>
    '/ Required method for Designer support - do not modify
    '/ the contents of this method with the code editor.
    '/ </summary>
    Private Sub InitializeComponent() 
        Me.toolStripTop = New System.Windows.Forms.ToolStrip()
        Me.newToolStripButton = New System.Windows.Forms.ToolStripButton()
        Me.mixToolStripButton = New System.Windows.Forms.ToolStripButton()
        Me.instanceToolStripButton = New System.Windows.Forms.ToolStripButton()
        Me.panelDisplayCulture = New System.Windows.Forms.Panel()
        Me.toolStripTop.SuspendLayout()
        Me.SuspendLayout()
        ' 
        ' toolStripTop
        ' 
        Me.toolStripTop.Cursor = System.Windows.Forms.Cursors.Default
        Me.toolStripTop.GripStyle = System.Windows.Forms.ToolStripGripStyle.Hidden
        Me.toolStripTop.Items.AddRange(New System.Windows.Forms.ToolStripItem() {Me.newToolStripButton, Me.mixToolStripButton, Me.instanceToolStripButton})
        Me.toolStripTop.Location = New System.Drawing.Point(0, 0)
        Me.toolStripTop.Name = "toolStripTop"
        Me.toolStripTop.Size = New System.Drawing.Size(377, 25)
        Me.toolStripTop.TabIndex = 15
        Me.toolStripTop.Visible = True
        ' 
        ' newToolStripButton
        ' 
        Me.newToolStripButton.Name = "newToolStripButton"
        Me.newToolStripButton.Text = "New culture"
        ' 
        ' mixToolStripButton
        ' 
        Me.mixToolStripButton.Name = "mixToolStripButton"
        Me.mixToolStripButton.Text = "Mix cultures"
        ' 
        ' instanceToolStripButton
        ' 
        Me.instanceToolStripButton.Name = "instanceToolStripButton"
        Me.instanceToolStripButton.Text = "New instance"
        ' 
        ' panelDisplayCulture
        ' 
        Me.panelDisplayCulture.Location = New System.Drawing.Point(4, 26)
        Me.panelDisplayCulture.Name = "panelDisplayCulture"
        Me.panelDisplayCulture.Size = New System.Drawing.Size(369, 374)
        Me.panelDisplayCulture.TabIndex = 16
        ' 
        ' CultureBuilder
        ' 
        Me.ClientSize = New System.Drawing.Size(377, 403)
        Me.Controls.Add(panelDisplayCulture)
        Me.Controls.Add(toolStripTop)
        Me.Name = "CultureBuilder"
        Me.Text = "Culture Builder"
        Me.toolStripTop.ResumeLayout(False)
        Me.toolStripTop.PerformLayout()
        Me.ResumeLayout(False)
        Me.PerformLayout()
    
    End Sub 'InitializeComponent
    #End Region
    
    
    Private Sub newToolStripButton_Click(ByVal sender As Object, ByVal e As System.EventArgs)  Handles newToolStripButton.Click
        Dim newCultureForm As NewCulture
        newCultureForm = New NewCulture(helper)
        newCultureForm.Show()
    End Sub 'newToolStripButton_Click
    
    
    Private Sub mixToolStripButton_Click(ByVal sender As Object, ByVal e As System.EventArgs)  Handles mixToolStripButton.Click
        Dim mixCulturesForm As MixCultures
        mixCulturesForm = New MixCultures(helper)
        mixCulturesForm.Show()
    End Sub 'mixToolStripButton_Click
    
    
    Private Sub instanceToolStripButton_Click(ByVal sender As Object, ByVal e As System.EventArgs)  Handles instanceToolStripButton.Click
        Dim newInstanceForm As NewInstance
        newInstanceForm = New NewInstance(helper)
        newInstanceForm.Show()
    End Sub 'instanceToolStripButton_Click
End Class 'CultureBuilder

File: 2005/Windows SDK NetFx 3.0/Applications/CultureSample/CS/CultureSample/CultureSample.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Globalization;

namespace Microsoft.Samples.Globalization.Culture
{
	/// <summary>
	/// Main window for the application.
	/// </summary>
	public class CultureBuilder : System.Windows.Forms.Form
	{
		#region Windows Form Designer declarations
		private System.Windows.Forms.ToolStrip toolStripTop;
		private System.Windows.Forms.ToolStripButton newToolStripButton;
		private System.Windows.Forms.ToolStripButton mixToolStripButton;
		private System.Windows.Forms.ToolStripButton instanceToolStripButton;
		private System.Windows.Forms.Panel panelDisplayCulture;
		#endregion

		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		//Helper to pass on to the children windows
		private CultureInfoHelper helper;

		public CultureBuilder()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();
			DisplayCulture display = new DisplayCulture();
			helper = new CultureInfoHelper(display);
			display.LoadComboBox(helper.GetCultures(CultureTypes.AllCultures));
			panelDisplayCulture.Controls.Add(display);	
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.toolStripTop = new System.Windows.Forms.ToolStrip();
			this.newToolStripButton = new System.Windows.Forms.ToolStripButton();
			this.mixToolStripButton = new System.Windows.Forms.ToolStripButton();
			this.instanceToolStripButton = new System.Windows.Forms.ToolStripButton();
			this.panelDisplayCulture = new System.Windows.Forms.Panel();
			this.toolStripTop.SuspendLayout();
			this.SuspendLayout();
// 
// toolStripTop
// 
			this.toolStripTop.Cursor = System.Windows.Forms.Cursors.Default;
			this.toolStripTop.GripStyle = System.Windows.Forms.ToolStripGripStyle.Hidden;
			this.toolStripTop.Items.AddRange(new System.Windows.Forms.ToolStripItem[] {
				this.newToolStripButton, this.mixToolStripButton, this.instanceToolStripButton
			});
			this.toolStripTop.Location = new System.Drawing.Point(0, 0);
			this.toolStripTop.Name = "toolStripTop";
			this.toolStripTop.Size = new System.Drawing.Size(377, 25);
			this.toolStripTop.TabIndex = 15;
			this.toolStripTop.Visible = true;
// 
// newToolStripButton
// 
			this.newToolStripButton.Name = "newToolStripButton";
			this.newToolStripButton.Text = "New culture";
			this.newToolStripButton.Click += new System.EventHandler(this.newToolStripButton_Click);
// 
// mixToolStripButton
// 
			this.mixToolStripButton.Name = "mixToolStripButton";
			this.mixToolStripButton.Text = "Mix cultures";
			this.mixToolStripButton.Click += new System.EventHandler(this.mixToolStripButton_Click);
// 
// instanceToolStripButton
// 
			this.instanceToolStripButton.Name = "instanceToolStripButton";
			this.instanceToolStripButton.Text = "New instance";
			this.instanceToolStripButton.Click += new System.EventHandler(this.instanceToolStripButton_Click);
// 
// panelDisplayCulture
// 
			this.panelDisplayCulture.Location = new System.Drawing.Point(4, 26);
			this.panelDisplayCulture.Name = "panelDisplayCulture";
			this.panelDisplayCulture.Size = new System.Drawing.Size(369, 374);
			this.panelDisplayCulture.TabIndex = 16;
// 
// CultureBuilder
// 
			this.ClientSize = new System.Drawing.Size(377, 403);
			this.Controls.Add(this.panelDisplayCulture);
			this.Controls.Add(this.toolStripTop);
			this.Name = "CultureBuilder";
			this.Text = "Culture Builder";
			this.toolStripTop.ResumeLayout(false);
			this.toolStripTop.PerformLayout();
			this.ResumeLayout(false);
			this.PerformLayout();
		}
		#endregion

		private void newToolStripButton_Click(object sender, System.EventArgs e)
		{
			new NewCulture(helper).Show();
		}

		private void mixToolStripButton_Click(object sender, System.EventArgs e)
		{
			new MixCultures(helper).Show();
		}

		private void instanceToolStripButton_Click(object sender, System.EventArgs e)
		{
			new NewInstance(helper).Show();
		}
	}
}
